### ISSUE TITLE
volatile. prefix on indirect access (ldind/stind) dropped and reported Full

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1434]
## Goal

Over-raise honesty bug for the #1356 burndown (fidelity-model class). A `volatile.`-prefixed indirect access (`volatile. ldind` / `volatile. stind`) is decompiled to a plain `*p` dereference that drops the acquire/release semantics, and fidelity is reported **`Full`** — the model cannot even distinguish the volatile read from a non-volatile one.

## The over-raise claim being narrowed

The importer faithfully captures the prefix — `IrImporter.cs:734` sets `volatilePrefix`, threaded onto `LoadIndirect`/`StoreIndirect` as `IsVolatile` (`IrImporter.cs:984/994/1009/1018`). But `IsVolatile` is then read by **nothing**: the printer's `DerefLoad`/`Deref` (`CSharpPrinter.cs:~1653`) ignores it and emits `*p`, and the fidelity model (`IrFunction.Fidelity` in `IrNodes.cs`, `FidelityRemarks.Collect` in `FidelityRemarks.cs`) has no `IsVolatile` clause. (Grep: outside the importer and `Describe()`, `IsVolatile` has no readers.) A volatile pointer/ref read has no plain-C# spelling, so emitting `*p` at `Full` is misleading.

(Note: `volatile.` on a *field* access is faithful — csc only emits it when the field is declared `volatile`, so the volatility lives on the declaration. The gap is specifically the indirect `ldind`/`stind` form.)

## Minimal counterexample (synthetic IR fixture — csc cannot emit `volatile.` on a bare deref)

```csharp
var intType = TypeRef.CoreLib("System", "Int32");
var intPtr  = TypeRef.Pointer(intType);
IrFunction Build(bool isVolatile)
{
    var load = new LoadIndirect(intType, new LoadArgument(0, "p", intPtr)) { IsVolatile = isVolatile };
    var c = new BlockContainer(); var b = new Block(0); c.Add(b); b.Add(new Return(load));
    var sig = new MethodSignature(intType, [new Parameter("p", intPtr)], HasThis: false, GenericParameterCount: 0);
    return new IrFunction("ReadVolatile", TypeRef.CoreLib("Synthetic","T"), sig, [], c);
}
```

Result (both grades `Full`, identical output — the near-miss negative is built in):

| `IsVolatile` | Fidelity | output |
| --- | --- | --- |
| `true` (`volatile. ldind`) | `Full` | `return *p;` |
| `false` (plain `ldind`) | `Full` | `return *p;` |

The two are indistinguishable, so a volatile read silently loses its memory-ordering guarantee at `Full`.

## Missing gate

No consumer of `IsVolatile` in the printer or the fidelity model. A `volatile.` indirect access has no faithful plain-C# spelling.

## First fix shape

Add an `IsVolatile`-true clause to the fidelity model (`IrFunction.Fidelity` / `FidelityRemarks.Collect`) so a `volatile. ldind`/`stind` caps at `Partial` with a remark (there is no faithful body spelling), **or** raise it at the printer to `Volatile.Read(ref …)` / `Volatile.Write(ref …)` where a `ref` spelling is recoverable. Field-access volatility stays `Full` (faithful on the declaration).

## Done signal

- Adversarial fixture: the `volatile. ldind`/`stind` indirect case is no longer reported `Full` with a bare `*p` (either `Partial`+remark, or a faithful `Volatile.Read/Write`).
- Positive canary: a plain (non-volatile) `ldind`/`stind` still prints `*p` at `Full`; a `volatile` *field* access stays `Full`.

## Notes

Confirmed: importer→printer→fidelity chain verified by source (importer sets the flag; nothing reads it) and by the synthetic IR fixture above (run through `CSharpPrinter.Print`). Reachable from any IL body carrying `volatile. ldind/stind` (hand-IL and some interop/marshalling stubs). Fidelity-honesty class: `Full` reported for output that drops observable behavior with no faithful spelling.
